internal/downloader/downloader.go

```diff
@@ -108,7 +109,19 @@ func NewBucketManager(buckets []cfg.BucketConfig) (*BucketManager, error) {
 	clients := make([]*minio.Client, 0, len(buckets))
 	bucketNames := make([]string, 0, len(buckets))
 	for _, bkt := range buckets {
-		client, err := minio.New(bkt.Host, &minio.Options{
+		host := bkt.Host
+		if bkt.ResolveIP {
+			hostname, port, err := net.SplitHostPort(bkt.Host)
+			if err != nil {
+				hostname = bkt.Host
+			}
+			addrs, err := net.LookupHost(hostname)
+			if err != nil {
+				return nil, fmt.Errorf("failing to lookup IP for %s: %w", bkt.Host, err)
+			}
+			host = net.JoinHostPort(addrs[0], port)
```

Comment:
What if the server goes away e.g. for maintenance? I would probably suggest contributing to minio-go the ability to specify a custom `Host` header.